src/matches.rs

```diff
@@ -263,19 +271,27 @@ fn rewrite_match_arm(
     };
 
     // Patterns
-    let pat_shape = match &arm.body.as_ref()?.kind {
+    let pat_shape = match &arm.body.as_ref().unknown_error()?.kind {
         ast::ExprKind::Block(_, Some(label)) => {
             // Some block with a label ` => 'label: {`
             // 7 = ` => : {`
             let label_len = label.ident.as_str().len();
-            shape.sub_width(7 + label_len)?.offset_left(pipe_offset)?
+            shape
+                .sub_width(7 + label_len)
+                .max_width_error(shape.width, arm.span)?
+                .offset_left(pipe_offset)
+                .max_width_error(shape.width, arm.span)?
```

Comment:
Unfortunate that we have to call `.max_width_error` twice here, but it makes sense. I think we should eventually chate `Shap::sub_width` and other methods to return `Result<Shape, RewriteError>`, but that can wait for a future PR.

Reply:
I think we can replace that with `.sub_width().and_then(|| offset_left()).max_width_error()` to avoid calling `.max_width_error` twice, but there might be a better approach.

Reply:
Personally, I think that would work well here!